apps/web/pages/_document.tsx

```diff
@@ -17,12 +17,17 @@ function setHeader(ctx: NextPageContext, name: string, value: string) {
 class MyDocument extends Document<Props> {
   static async getInitialProps(ctx: DocumentContext) {
     const { nonce } = csp(ctx.req || null, ctx.res || null);
-    if (!process.env.CSP_POLICY) {
-      setHeader(ctx, "x-csp", "not-opted-in");
-    } else if (!ctx.res?.getHeader("x-csp")) {
-      // If x-csp not set by gSSP, then it's initialPropsOnly
-      setHeader(ctx, "x-csp", "initialPropsOnly");
+    try {
+      if (!process.env.CSP_POLICY) {
+        setHeader(ctx, "x-csp", "not-opted-in");
+      } else if (!ctx.res?.getHeader("x-csp")) {
+        // If x-csp not set by gSSP, then it's initialPropsOnly
+        setHeader(ctx, "x-csp", "initialPropsOnly");
+      }
+    } catch (e) {
+      console.log((e as Error).message);
```

Comment:
I don't think we need this try-catch. `res.setHeader` can throw error which is already try catched by `setHeader` function. getHeader shouldn't throw any error